tiledb/sm/fragment/fragment_metadata.cc

```diff
@@ -974,52 +975,74 @@ std::string FragmentMetadata::encode_name(const std::string& name) const {
         percent_encoded_name << percent_encoding.at(c);
     }
 
-    return percent_encoded_name.str();
+    return {Status::Ok(), percent_encoded_name.str()};
   }
 
   assert(version_ > 8);
   const auto iter = idx_map_.find(name);
-  if (iter == idx_map_.end())
-    LOG_FATAL("Name " + name + " not in idx_map_");
+  if (iter == idx_map_.end()) {
+    auto err = "Name " + name + " not in idx_map_";
+    LOG_ERROR(err);
```

Comment:
Since we are returning a status object up the function stack now, I think we can remove the `LOG_ERROR` here. The status can be logged in a higher level function now.

Reply:
Makes sense. Do we know if the caller code actually logs stuff at this point if status turns out to be error? It's hard for me to understand now if removing the LOG_ERRORs from here will create blindspots because caller code relied on LOG_FATAL to write to logs before this PR.

Reply:
Great question, as far as I can tell this will propagate correctly back to the c-api for the user to get the error status.

Reply:
Good then, should I proceed to removing them?

Reply:
Yes